### PR TITLE
ci: init visual testing with loki

### PR DESCRIPTION
**Pull request by vincentaudebert**
## Type

- Feature

#### What is expected?

Visual testing in pipeline

#### The following changes where made:

(Describe what you did)

1. Adds loki to visual test components based on storybook stories

### CODE REVIEW

**Comment by vincentaudebert** [comment]
Note to myself: reset matrix on test + build ---- node: ["14", "16", "17"]

File: src/components/Placeholder/__stories__/index.stories.tsx [hunk 1]
```diff
@@ -2,6 +2,8 @@ import { Meta, Story } from '@storybook/react'
 import React, { ComponentProps } from 'react'
 import Placeholder from '..'
 
+console.log(Math.random(), Math.random(), Math.random())
```
[the remark this answers is not in the document]
Comment (vincentaudebert) [reply src/components/Placeholder/__stories__/index.stories.tsx]: Indeed 😅 fixed

**Review by adriengibrat — APPROVED** [review]

**Review by chambo-e — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -58,23 +58,65 @@ jobs:
...
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2.3.5
+      - name: Use Node.js
+        uses: actions/setup-node@v2.4.1
+        with:
+          node-version: 17
+      - run: yarn install
+      - run: yarn run build:storybook-visual
+      - uses: actions/cache@v2
```
[the remark this answers is not in the document]
Comment (vincentaudebert) [reply .github/workflows/ci.yml:88]: @chambo-e No this is caching the build to be used later on in the "visual" job

Artefacts are stored there

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -58,23 +58,65 @@ jobs:
...
-    - run: yarn install
-    - run: yarn run build
+      - uses: actions/checkout@v2.3.5
+      - name: Use Node.js
+        uses: actions/setup-node@v2.4.1
+        with:
+          node-version: ${{ matrix.node }}
+      - run: yarn install
+      - run: yarn run build
+  build-storybook-visual:
```
[the remark this answers is not in the document]
Comment (vincentaudebert) [reply .github/workflows/ci.yml:78]: Not sure I get it? This is the first step for visual testing CI. (if you check loki, they do build storybook then they run the visual testing. We need to build first to target local build with loki

[the remark this answers is not in the document]
Comment (chambo-e) [reply .github/workflows/ci.yml:78]: This workflow is performed onto every pull requests and every push on main

As we only want this on pull requests we could have a dedicated Visual Testing pipeline subscribed only to PR events :)

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (chambo-e) [reply .github/workflows/ci.yml:88]: The build is an artefact no ?

**Review by matthprost — CHANGES_REQUESTED** [review]
Isn't github repo gonna be super heavy to clone with all those pictures ? Is this the only way to keep them locally ? Cc @chambo-e

**Comment by chambo-e** [comment]
> Isn't github repo gonna be super heavy to clone with all those pictures ? Is this the only way to keep them locally ? Cc @chambo-e

@matthprost there is only 3~Mb of png, I think we can handle it for now :)

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vincentaudebert) [reply .github/workflows/ci.yml:88]: @chambo-e i used this to decide, i think it's pretty complete

https://levelup.gitconnected.com/github-actions-how-to-share-data-between-jobs-fc1547defc3e

File: .github/workflows/ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (vincentaudebert) [reply .github/workflows/ci.yml:78]: @chambo-e good point. I'm on it

File: package.json [hunk 4]
```diff
@@ -48,7 +49,9 @@
     "test:unit:update": "yarn run test:unit --updateSnapshot",
     "test:a11y": "LC_ALL=en_US.UTF-8 jest --config .jest/a11y.config.ts",
     "prepare": "husky install",
-    "size": "yarn run build && size-limit"
+    "size": "yarn run build && size-limit",
+    "visual:update": "yarn run build:storybook-visual && yarn loki update",
+    "visual:test": "yarn loki test"
```
Comment (chambo-e) [inline package.json]:
```suggestion
    "test:visual:update": "yarn run build:storybook-visual && yarn loki update",
    "test:visual": "yarn loki test"
```

To stay consistent with other test commands

File: package.json [hunk 5]
```diff
@@ -145,7 +148,9 @@
     "jest": "27.5.1",
     "jest-axe": "5.0.1",
     "jest-junit": "13.0.0",
+    "jest-mock-random": "^1.1.1",
```
Comment (chambo-e) [inline package.json]:
```suggestion
    "jest-mock-random": "1.1.1",
```

File: package.json [hunk 6]
```diff
@@ -145,7 +148,9 @@
...
     "lint-staged": "12.3.4",
+    "loki": "^0.28.1",
```
Comment (chambo-e) [inline package.json]:
```suggestion
    "loki": "0.28.1",
```

**Review by chambo-e — CHANGES_REQUESTED** [review]

**Pull request merged**